code/modules/power/groundmap_geothermal.dm

```diff
@@ -89,8 +93,8 @@
 
 /obj/machinery/power/geothermal/process()
 	if(corrupted && corruption_on)
-		if(length(GLOB.humans_by_zlevel["2"]) > 0.2 * length(GLOB.alive_human_list))
-			SSpoints.add_psy_points("[corrupted]", GENERATOR_PSYCH_POINT_OUTPUT / GLOB.geothermal_generator_ammount)
+		if(length(GLOB.humans_by_zlevel["2"]) > 0.2 * length(GLOB.alive_human_list_faction[FACTION_TERRAGOV]))
+			SSpoints.add_psy_points(corrupted, GENERATOR_PSYCH_POINT_OUTPUT / GLOB.xeno_generators_by_hive[corrupted])
```

Comment:
This is vulnerable to issues if another hive corrupts the gen after it is decorrupted (e.g. corrupted hive vs. normal hive), as the list of generators per hive does not ever change and thus can cause div-by-0 errors in this case.